packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -39,28 +40,44 @@ export interface MigrationFile {
 
 type MIGRATION_STEP = typeof MIGRATION_STEPS[number]
 
+const flags = {
+  hasCustomIntegrationFolder: false,
+  hasCustomIntegrationTestFiles: false,
+
+  hasCustomComponentFolder: false,
+  hasCustomComponentTestFiles: false,
+
+  hasCustomSupportFile: false,
+  hasComponentTesting: true,
+  hasE2ESpec: true,
+  hasPluginsFile: true,
+} as const
+
 export class MigrationDataSource {
   private _config: OldCypressConfig | null = null
   private _step: MIGRATION_STEP = 'renameAuto'
   filteredSteps: MIGRATION_STEP[] = MIGRATION_STEPS.filter(() => true)
 
-  hasCustomIntegrationFolder: boolean = false
-  hasCustomIntegrationTestFiles: boolean = false
+  hasCustomIntegrationFolder: boolean = flags.hasCustomIntegrationFolder
```

Comment:
I think the type is inferred here, but I am guessing declaring the types here is intentionally explicit to make the type clear?

Reply:
exactly

Reply:
yes!